### PR TITLE
Initial v0.1 Implementation

### PR DESCRIPTION
**Pull request by jefflinse**
### SUMMARY

Implement initial action for tagging new releases based on PR tags and release notes.

### RELEASE NOTES

Initial v0.1 implementation

Adds support for
- 'validation' mode: ensure that an appropriate release label is set on the PR, and ensure that release notes are present, if they are required.
- 'bump' mode: bump the version based on the type of release and tag a new release with the available release notes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "lint": "eslint .",
```
Comment (Setheck) [inline package.json:7]: you should probably run
`npm install eslint --save-dev`
and checkin the package.json

File: index.js [hunk 2]
```diff
@@ -1,11 +1,134 @@
...
+const { getConfig } = require('./config');
+const { extractPRNumber, fetchPR, getReleaseType, getReleaseNotes } = require("./pr");
+const { octokit } = require("./octokit");
 
 async function run() {
     try {
-        core.info(`Hello, World!`);
-    } catch (error) {
-        core.setFailed(error.message);
+        var config = getConfig()
```
Comment (Setheck) [inline index.js:10]: i believe var is old scool and `let` is now for scoped variables

File: index.js [hunk 3]
```diff
@@ -1,11 +1,134 @@
...
+
+    core.info(`current version: ${config.v}${currentVersion}`)
+    core.info(`next version: ${config.v}${newVersion}`)
+    core.info(`release notes:\n${releaseNotes}`)
+
+    core.setOutput('oldversion', `${config.v}${currentVersion}`)
+    core.setOutput('version', `${config.v}${newVersion}`)
+    core.setOutput('releasenotes', releaseNotes)
+
+    return
```
Comment (Setheck) [inline index.js:56]: this return is superfluous

File: index.js [hunk 4]
```diff
@@ -1,11 +1,134 @@
...
+    const currentVersion = await getCurrentVersion()
+    const newVersion = semver.inc(currentVersion, releaseType)
+
+    core.info(`current version: ${config.v}${currentVersion}`)
+    core.info(`next version: ${config.v}${newVersion}`)
+    core.info(`release notes:\n${releaseNotes}`)
+
+    core.setOutput('oldversion', `${config.v}${currentVersion}`)
+    core.setOutput('version', `${config.v}${newVersion}`)
+    core.setOutput('releasenotes', releaseNotes)
```
Comment (Setheck) [inline index.js:54]: should there be spaces?
'old version'
'release notes'

File: index.js (hunk 2 above)
Comment (jefflinse) [reply index.js]: Yep. Been a while since I've written JS.

File: index.js (hunk 4 above)
Comment (jefflinse) [reply index.js]: These are output variables to be accessed and used by other steps in a job, so they cannot contain spaces. Under the hood, inputs/outputs are just special env vars, and spaces are converted to underscores, but I prefer to be deliberate. I've updated these to use dashes to make them easier to read.

File: package.json (hunk 1 above)
Comment (jefflinse) [reply package.json:7]: The dev dependency is already there, it was part of the template I used to create the project.

File: config.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    var releaseLabels = {};
+    releaseLabels[core.getInput('major-label') || 'major release'] = 'major';
+    releaseLabels[core.getInput('minor-label') || 'minor release'] = 'minor';
+    releaseLabels[core.getInput('patch-label') || 'patch release'] = 'patch';
+
+    return {
+        mode: mode,
+        releaseLabels: releaseLabels,
+        releaseNotesRegex: new RegExp(`${releaseNotesPrefix}([\\s\\S]*)${releaseNotesSuffix}`),
```
Comment (siggimoo) [inline config.js:21]: nit: `[^]` is more concise than `[\\s\\S]`.

**Pull request merged**